server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3105,9 +3109,10 @@ private Pair<List<ServiceOfferingJoinVO>, Integer> searchForServiceOfferingsInte
         Integer cpuSpeed = cmd.getCpuSpeed();
         Boolean encryptRoot = cmd.getEncryptRoot();
 
+        caller = _accountMgr.finalizeOwner(caller, accountName, domainId, projectId);
```

Comment:
should this be `owner` ?
`caller` should be a final variable which cannot be modified.